src/sds_data_model/dataframe.py

```diff
@@ -0,0 +1,254 @@
+from typing_extensions import Self
+from pathlib import Path
+from inspect import ismethod, signature
+
+from dataclasses import dataclass
+from typing_extensions import Self
+from pyspark.pandas import read_excel
+from pyspark.sql import DataFrame as SparkDataFrame
+
+from pyspark.pandas import DataFrame, read_excel
+from pyspark.sql import SparkSession
+from pyspark_vector_files import read_vector_files
+from pyspark_vector_files.gpkg import read_gpkg
+
+from pyspark.sql import SparkSession
+
+from sds_data_model._vector import _get_metadata, _get_name
+from sds_data_model.metadata import Metadata
+from sds_data_model._dataframe import _to_zarr_region, _create_dummy_dataset
+
+from pandas import DataFrame
+from functools import partial
+
+spark = SparkSession.getActiveSession()
+
+# create logger and set level to info
+logger = getLogger("sds")
+logger.setLevel(INFO)
+
+# create stream handler and set level to info
+stream_handler = StreamHandler()
+stream_handler.setLevel(INFO)
+
+# create formatter
+formatter = Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+# add formatter to stream handler
+stream_handler.setFormatter(formatter)
+
+# add stream handler to logger
+logger.addHandler(stream_handler)
+
+# DataFrameWrapper is an upper bound for _DataFrameWrapper.
+# Specifying bound means that _DataFrameWrapper will only be
+# DataFrameWrapper or one of its subclasses.
+_DataFrameWrapper = TypeVar("_DataFrameWrapper", bound="DataFrameWrapper")
+
+
+@dataclass
+class DataFrameWrapper:
+    name: str
+    data: SparkDataFrame
+    metadata: Optional[Metadata]
+    #graph: Optional[DiGraph]
+
+    @classmethod
+    def from_files(
+            cls: Type[_DataFrameWrapper],
+            data_path: str,
+            metadata_path: Optional[str] = None,
+            metadata_kwargs: Optional[Dict[str, Any]] = None,
+            name: Optional[str] = None,
+            read_file_kwargs: Optional[Dict[str, Any]] = None,
+            spark: Optional[SparkSession] = None
+            # optional spark session argument (_spark )
+    ) -> _DataFrameWrapper:
+        """Reads in data with a range of file types, and converts it to a spark dataframe,
+        wrapped with associated metadata
+        
+        Examples:
+        >>> from sds_data_model.dataframe import DataFrameWrapper
+        >>> wrapped_shp = DataFrameWrapper.from_files(name = "National parks",                               
+                                        data_path="/dbfs/mnt/base/unrestricted/source_defra_data_services_platform/dataset_national_parks/format_SHP_national_parks/LATEST_national_parks/",
+                                        read_file_kwargs = {'suffix':'.shp',
+                                                            'ideal_chunk_size':1000},
+                                        metadata_path =  "https://ckan.publishing.service.gov.uk/harvest/object/656c07d1-67b3-4bdb-8ab3-75e118a7cf14"
+                                       )
```

Comment:
You don't need the `ideal_chunk_size`, it's only relevant if you're using the rows concurrency strategy, which you aren't, and it's 1,000 by default anyway.

```suggestion
        >>> wrapped_shp = DataFrameWrapper.from_files(name = "National parks",                               
                                        data_path="/dbfs/mnt/base/unrestricted/source_defra_data_services_platform/dataset_national_parks/format_SHP_national_parks/LATEST_national_parks/",
                                        read_file_kwargs = {'suffix':'.shp'},
                                        metadata_path =  "https://ckan.publishing.service.gov.uk/harvest/object/656c07d1-67b3-4bdb-8ab3-75e118a7cf14"
                                       )
```